### PR TITLE
Add XPATH scraping possibility

### LINKED ISSUE #22: Create 'Simple'_WEB_COLLECTOR
Create a new collector that just uses trafilatura to parse the website and create a new item if there is a diff to last parsed.

- [ ] with the option of parsing specific text parts, e.g. with Xpath
- [ ] Add test to example page: https://www.wienernetze.at/stromversorgung

### PR DESCRIPTION
**Pull request by not4Pedro**
Resolves #22 

There are bugs found in the Simple Web Collector. Scraping some websites throws errors in the Core.

Example: https://www.bbc.com/news/live/world-middle-east-67906496
Error: [Core] [ERROR] - Updating status for 3b29d020-3e13-48dd-96ec-cf2f10741761 with error message string indices must be integers, not 'str'.

The place where it breaks is in [sanitation](https://github.com/taranis-ai/taranis-ai/blob/1e6bbe7849bfad9e3de9e9e09fbcf574319db6bf/src/worker/worker/collectors/base_collector.py#L77).

## Summary by CodeRabbit

- **New Features**
	- Enhanced web collector capabilities with XPath support for targeted content extraction.

- **Refactor**
	- Refactored web scraping collectors to inherit from a new base class, optimizing code reuse and functionality.
	- Updated method signatures with explicit parameter types for better clarity and type checking.

- **Bug Fixes**
	- Fixed hashing logic for improved data integrity.

- **Documentation**
	- Renamed and reorganized methods for better understanding and maintenance of the codebase.

### CODE REVIEW

**Comment by not4Pedro** [comment]
# /html/body/div[3]/main/div[1]/div/div[3]/div/div/section/div/div[2]/div/div/div/p[2]  https://www.wienernetze.at/stromversorgung

**Comment by not4Pedro** [comment]
@coderabbitai review

File: src/core/pyproject.toml [hunk 1]
```diff
@@ -33,7 +33,8 @@ dependencies = [
     "requests",
     "authlib",
     "sseclient",
-    "PyYAML==5.3.1"
+    "PyYAML==5.3.1",
+    "lxml.html"
```
Comment (b3n4kh) [inline src/core/pyproject.toml:37]: I think this should rather be added to src/worker/pyproject.toml

File: src/core/pyproject.toml [hunk 2]
```diff
@@ -54,7 +55,7 @@ fallback_version = "0.0.0"
 root = "../.."
 
 [project.optional-dependencies]
-dev = ["black", "flake8", "pytest", "pytest-celery", "sqlalchemy-stubs", "schemathesis", "build", "wheel", "setuptools", "setuptools_scm"]
+dev = ["black", "flake8", "pytest", "pytest-celery", "sqlalchemy-stubs", "schemathesis", "build", "wheel", "setuptools", "setuptools_scm", "lxml.html"]
```
Comment (b3n4kh) [inline src/core/pyproject.toml:58]: Since it as a decency needed during runtime, it should not be part of the "optional-dependencies"

**Review by b3n4kh — CHANGES_REQUESTED** [review]
Please move the dependency from core to worker

**Review by b3n4kh — APPROVED** [review]

**Pull request merged**